packages/@aws-cdk/aws-iam/lib/managed-policy.ts

```diff
@@ -1,22 +1,116 @@
+   * @attribute
    */
   readonly managedPolicyArn: string;
 }
 
+export interface ManagedPolicyProps {
+  /**
+   * The name of the managed policy. If you specify multiple policies for an entity,
+   * specify unique names. For example, if you specify a list of policies for
+   * an IAM role, each policy must have a unique name.
+   *
+   * @default - Uses the logical ID of the policy resource, which is ensured
+   * to be unique within the stack.
+   */
+  readonly managedPolicyName?: string;
+
+  /**
+   * A description of the managed policy. Typically used to store information about the
+   * permissions defined in the policy. For example, "Grants access to production DynamoDB tables."
+   * The policy description is immutable. After a value is assigned, it cannot be changed.
+   *
+   * @default - empty
+   */
+  readonly description?: string;
+
+  /**
+   * The path for the policy. This parameter allows (through its regex pattern) a string of characters
+   * consisting of either a forward slash (/) by itself or a string that must begin and end with forward slashes.
+   * In addition, it can contain any ASCII character from the ! (\u0021) through the DEL character (\u007F),
+   * including most punctuation characters, digits, and upper and lowercased letters.
+   *
+   * For more information about paths, see IAM Identifiers in the IAM User Guide.
+   *
+   * @default - "/"
+   */
+  readonly path?: string;
+
+  /**
+   * Users to attach this policy to.
+   * You can also use `attachToUser(user)` to attach this policy to a user.
+   *
+   * @default - No users.
+   */
+  readonly users?: IUser[];
+
+  /**
+   * Roles to attach this policy to.
+   * You can also use `attachToRole(role)` to attach this policy to a role.
+   *
+   * @default - No roles.
+   */
+  readonly roles?: IRole[];
+
+  /**
+   * Groups to attach this policy to.
+   * You can also use `attachToGroup(group)` to attach this policy to a group.
+   *
+   * @default - No groups.
+   */
+  readonly groups?: IGroup[];
+
+  /**
+   * Initial set of permissions to add to this policy document.
+   * You can also use `addPermission(statement)` to add permissions later.
+   *
+   * @default - No statements.
+   */
+  readonly statements?: PolicyStatement[];
+}
+
 /**
  * Managed policy
  *
- * This class is an incomplete placeholder class, and exists only to get access
- * to AWS Managed policies.
  */
-export class ManagedPolicy {
+export class ManagedPolicy extends Resource implements IManagedPolicy {
+  /**
+   * Construct a customer managed policy from the managedPolicyName
+   *
+   * For this managed policy, you only need to know the name to be able to use it.
+   *
+   */
+  public static fromManagedPolicyName(scope: Construct, id: string, managedPolicyName: string): IManagedPolicy {
+    class Import extends Resource implements IManagedPolicy {
+      public readonly managedPolicyArn = Lazy.stringValue({
+        produce(ctx: IResolveContext) {
```

Comment:
Why does this have to be a lazy string and not just:

```ts
const stack = Stack.of(scope);

class Import extends Resource implements IManagedPolicy {
  public readonly managedPolicyArn = stack.formatArn({...});

  ...
}
```

Reply:
I used the Policy class as a reference. If we want to change stuff I can do that. For the path thing, I think you're right. I'll update it

Reply:
> Doesn't the ARN consists of the path plus the name? `path` is not even a parameter here.

So for the import class I'm not sure if it needs to be a separate parameter on this function or if you can just pass the whole thing in as the managedPolicyName. Similarly to how fromAwsManagedPolicyName accepts the service-role prefix as part of the managedPolicyName.

Reply:
Looking at other Principals that can have a path, the Group class uses the ARN for lookup with a fromGroupArn method, the User class doesn't have a lookup. Perhaps we should make this lookup fromPolicyArn so that it's consistent with Group?

Reply:
I guess not